examples/granger_causality.py

```diff
@@ -0,0 +1,405 @@
+#
+# where: :math:`A` and :math:`B` are the seeds and targets, respectively;
+# :math:`f` is a given frequency; :math:`\boldsymbol{H}` is the spectral
+# transfer function; :math:`\boldsymbol{\Sigma}` is the innovations form
+# residuals' covariance matrix of the state-space model; :math:`\boldsymbol{S}`
+# is :math:`\boldsymbol{\Sigma}` transformed by :math:`\boldsymbol{H}`; and
+# :math:`\boldsymbol{\Sigma}_{IJ \lvert K} := \boldsymbol{\Sigma}_{IJ} -
+# \boldsymbol{\Sigma}_{IK} \boldsymbol{\Sigma}_{KK}^{-1}
+# \boldsymbol{\Sigma}_{KJ}`, representing a partial covariance matrix. The same
+# principles apply as before: a numerator greater than the denominator means
+# that information from the seed signals aids the prediction of activity in the
+# target signals, leading to a Granger score > 0.
+#
+# There are several benefits to a state-space approach for computing GC:
+# compared to traditional autoregressive-based approaches, the use of
+# state-space models offers reduced statistical bias and increased statistical
+# power; furthermore, the dimensionality reduction offered by the multivariate
+# nature of the approach can aid in the interpretability and subsequent
+# analysis of the results.
+#
+# To demonstrate the use of GC for estimating directed connectivity, we start
+# by loading some example MEG data and dividing it into two-second-long epochs.
+
+# %%
+
+raw = mne.io.read_raw_ctf(data_path() / 'SubjectCMC.ds')
+raw.pick('mag')
+raw.crop(50., 110.).load_data()
+raw.notch_filter(50)
+raw.resample(100)
+
+epochs = mne.make_fixed_length_epochs(raw, duration=2.0).load_data()
+
+###############################################################################
+# We will focus on connectivity between sensors over the parietal and occipital
+# cortices, with 20 parietal sensors designated as group A, and 20 occipital
+# sensors designated as group B.
+
+# %%
+
+# parietal sensors
+signals_a = [idx for idx, ch_info in enumerate(epochs.info['chs']) if
+             ch_info['ch_name'][2] == 'P']
+# occipital sensors
+signals_b = [idx for idx, ch_info in enumerate(epochs.info['chs']) if
+             ch_info['ch_name'][2] == 'O']
+
+# UNTIL RAGGED INDICES SUPPORTED
+min_n_chs = min(len(signals_a), len(signals_b))
+signals_a = signals_a[:min_n_chs]
+signals_b = signals_b[:min_n_chs]
+
+indices_ab = (np.array(signals_a), np.array(signals_b))  # A => B
+indices_ba = (np.array(signals_b), np.array(signals_a))  # B => A
+
+signals_a_names = [epochs.info['ch_names'][idx] for idx in signals_a]
+signals_b_names = [epochs.info['ch_names'][idx] for idx in signals_b]
+
+# compute Granger causality
+gc_ab = spectral_connectivity_epochs(
+    epochs, method=['gc'], indices=indices_ab, fmin=5, fmax=30,
+    rank=(np.array([5]), np.array([5])), gc_n_lags=20)  # A => B
+gc_ba = spectral_connectivity_epochs(
+    epochs, method=['gc'], indices=indices_ba, fmin=5, fmax=30,
+    rank=(np.array([5]), np.array([5])), gc_n_lags=20)  # B => A
+freqs = gc_ab.freqs
```

Comment:
Can you write a note here explaining the intuition of `rank` and `gc_n_lags`? I know they are in the docstring, but perhaps for a user that is used to the original API of `spectral_connectivity_epochs`, a short note here might help them understand how to set the values as well.

Reply:
And perhaps the same in the `mic_mim.py` example

Reply:
For `gc_n_lags`, I explain this in the section below (applies only to GC, not to MIC/MIM): https://github.com/tsbinns/mne-connectivity/blob/dae09a5ff3a1bd5caf848d8ab711d1f3887280ad/examples/granger_causality.py#L309-L319

As for `rank`, this applies to GC and MIC/MIM, so I explain this in the section below for the GC example: https://github.com/tsbinns/mne-connectivity/blob/dae09a5ff3a1bd5caf848d8ab711d1f3887280ad/examples/granger_causality.py#L342-L382
... and for the MIC/MIM example: https://github.com/tsbinns/mne-connectivity/blob/dae09a5ff3a1bd5caf848d8ab711d1f3887280ad/examples/mic_mim.py#L293-L326